app/src/main/java/com/example/tasky/feature_agenda/presentation/agenda_screen/AgendaViewModel.kt

```diff
@@ -0,0 +1,119 @@
+import com.example.tasky.feature_authentication.domain.util.UserPreferences
+import com.example.tasky.util.Result
+import com.vanpra.composematerialdialogs.MaterialDialogState
+import dagger.hilt.android.lifecycle.HiltViewModel
+import kotlinx.coroutines.channels.Channel
+import kotlinx.coroutines.flow.MutableStateFlow
+import kotlinx.coroutines.flow.asStateFlow
+import kotlinx.coroutines.flow.first
+import kotlinx.coroutines.flow.receiveAsFlow
+import kotlinx.coroutines.flow.update
+import kotlinx.coroutines.launch
+import javax.inject.Inject
+
+@HiltViewModel
+class AgendaViewModel @Inject constructor(
+    private val useCases: AgendaUseCases,
+    private val repositories: AgendaRepositories,
+    private val userPreferences: UserPreferences
+): ViewModel() {
+
+    private val _state = MutableStateFlow(AgendaState())
+    val state = _state.asStateFlow()
+
+    var dateDialogState by mutableStateOf(MaterialDialogState())
+
+    var username by mutableStateOf("")
+        private set
+
+    private val resultChannel = Channel<Result<Unit>>()
+    val logoutResult = resultChannel.receiveAsFlow()
+
+    init {
+        val user = userPreferences.getAuthenticatedUser()
+        user?.let {
+            val names = it.fullName.split(" ")
+            username = when(names.size) {
+                1 -> {
+                    "${names[0][0]}${names[0][1]}"
+                }
+
+                2 -> {
+                    "${names[0][0]}${names[1][0]}"
+                }
+
+                else -> ""
+            }
+        }
+        getAgenda()
+    }
+
+    fun onEvent(event: AgendaEvent) {
+
+        when(event) {
+            is AgendaEvent.Logout -> {
+                logout()
+            }
+            is AgendaEvent.SelectDate -> {
+                _state.update {
+                    it.copy(
+                        currentMonth = event.date.month,
+                        daysList = generateNextDays(event.date),
+                        currentDate = event.date
+                    )
+                }
+                getAgenda()
+            }
+            is AgendaEvent.SwipeToRefresh -> {
+                getRemoteAgenda()
+            }
+
+            is AgendaEvent.SelectDayIndex -> {
+                _state.update {
+                    it.copy(selectedDayIndex = event.index)
+                }
+                getAgenda()
+            }
+
+            is AgendaEvent.ToggleLogoutBtn -> {
+                _state.update {
+                    it.copy(
+                        showLogoutOption = event.showOption
+                    )
+                }
+            }
+        }
+    }
+
+    private fun getAgenda() {
+        viewModelScope.launch {
+            val agendaItems = repositories.agendaRepository.getAgendaForSpecificDay(state.value.currentDate).first()
+            _state.update {
+                it.copy(itemList = agendaItems)
+            }
+        }
```

Comment:
Why not make this reactive? Right now, you only fetch the agenda once, but actually collecting the flow would give you the advantage that the UI is always up-to-date. For example, when the background worker syncs and a new item gets added which was added on another device, it would automatically show up in the UI, since the background worker saved it in the DB which triggers the flow